src/components/ArticleList.js

```diff
@@ -1,24 +1,33 @@
-'use strict';
+"use strict";
 
-import ArticlePreview from './ArticlePreview';
-import React from 'react';
+import ArticlePreview from "./ArticlePreview";
+import React from "react";
 
-const ArticleList = props => {
-  if (!props.articles) {
-    return <div className="article-preview">Loading...</div>;
+class ArticleList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.articles = props.articles;
   }
 
-  if (props.articles.length === 0) {
-    return <div className="article-preview">No articles are here... yet.</div>;
-  }
+  render() {
+    if (!this.articles) {
+      return <div className="article-preview">Loading...</div>;
+    }
+
+    if (this.articles.length === 0) {
+      return (
+        <div className="article-preview">No articles are here... yet.</div>
+      );
+    }
 
-  return (
-    <div>
-      {props.articles.map(article => {
-        return <ArticlePreview article={article} key={article.slug} />;
-      })}
-    </div>
-  );
-};
+    return (
+      <div>
+        {this.articles.map(article => {
```

Comment:
So as far as naming it seems that what `ArticleList` component is should be more `ArticlesContainer` component since it handles the undefined and empty state and redirect the list rendering to a child component `ArticlePreview`